cli/command/container/run.go

```diff
@@ -186,7 +186,11 @@ func runContainer(ctx context.Context, dockerCli command.Cli, runOpts *runOption
 		defer closeFn()
 	}
 
-	statusChan := waitExitOrRemoved(ctx, apiClient, containerID, copts.autoRemove)
+	// New context here because we don't to cancel waiting on container exit/remove
+	// when we cancel attach, etc.
+	statusCtx, cancelStatusCtx := context.WithCancel(context.Background())
```

Comment:
Good point on both; is your thinking here to preserve the original context for tracing purposes (OTEL)? Given that go1.20 is EOL, perhaps it's ok to start ignoring it (although not sure if this code has made its way into other projects that didn't upgrade yet).

It's an interesting point though; `context.Background()` used to be the (IIUC) "correct" approach for spinning up something that's decoupled from the main context (w.r.t. cancelation at least). `WithoutCancel` does provide similar features (but also may be slightly less clear on intent).

Perhaps also something to consider discussing in our OTEL sync call; have a common approach for this (which _could_ even be a utility that's in a shared OTEL package, although there may be a risk on over-engineering things there.

Reply:
The issue is IMO in the `context.Context` itself lumping two distinct features (arbitrary key-value store AND passing cancellation/deadline around).

In this case, the intent is to create a context that's not immediately canceled when the parent context is cancelled (so effectively ignoring the parent context cancellation) while still persisting the key-value feature (because that's how OTEL passes the traces around).

Reply:
Yup, 💯 . And to _some extent_ it would then make sense to have a utility in some OTEL-related package, but then again "key-value" could be there for other purposes (like containerd's context-logger), so ... maybe not.

Reply:
Good catch, thanks!